ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -543,18 +543,25 @@ private IndexValue findKey(StoreKey key, FileSpan fileSpan, IndexEntryType type,
       for (Map.Entry<Offset, IndexSegment> entry : segmentsMapToSearch.entrySet()) {
         segmentsSearched++;
         logger.trace("Index : {} searching index with start offset {}", dataDir, entry.getKey());
-        IndexValue value = entry.getValue().find(key);
-        if (value != null) {
-          logger.trace("Index : {} found value offset {} size {} ttl {}", dataDir, value.getOffset(), value.getSize(),
-              value.getExpiresAtMs());
-          if (type.equals(IndexEntryType.ANY)) {
-            retValue = value;
-            break;
-          } else if (type.equals(IndexEntryType.DELETE) && value.isFlagSet(IndexValue.Flags.Delete_Index)) {
-            retValue = value;
-            break;
-          } else if (type.equals(IndexEntryType.PUT) && !value.isFlagSet(IndexValue.Flags.Delete_Index)) {
-            retValue = value;
+        NavigableSet<IndexValue> values = entry.getValue().find(key);
+        if (values != null) {
```

Comment:
Do the tests cover this method well? Cases like the key being present in multiple segments with the intended type being in an older segment and so on?

Reply:
Yeah - that was always covered. All cases are covered
1. Put only present
2. Delete only present
3. Put and Delete both present in the same index segment
4. Put and Delete in different index segments